planner/core/rule_decorrelate.go

```diff
@@ -189,6 +189,13 @@ func (s *decorrelateSolver) optimize(ctx context.Context, p LogicalPlan, opt *lo
 			}
 			appendRemoveProjTraceStep(apply, proj, opt)
 			return s.optimize(ctx, p, opt)
+		} else if li, ok := innerPlan.(*LogicalLimit); ok {
+			if li.Offset == 0 {
```

Comment:
Add some comments here about why we could apply the rule and why we should only consider the `offset=0`

Reply:
Add some comments

Reply:
> We also need to add the test mentioned in the issue.

add this test which mentioned.